### ISSUE TITLE
escaping banner makes bar show up in middle of screen``

### ISSUE THREAD

**Issue by InvenTorrey** [fcaviggia/classification-banner#37]
the bar appears correctly initially, but if you escape it to make it disappear, it will reappear in the middle of the screen

**Comment by fcaviggia** [comment]
What OS, Xorg/Wayland, display manager, etc.  Need more information to help here.

**Comment by djtorrey** [comment]
Ubuntu Budgie 18.10 is what I was working with, though I will be pushing it to Centos7.  I'll check it out after pushing it first

**Comment by fcaviggia** [comment]
I have never got it to work with Mir due to the trying to determine resolution - I've had luck with every other window manager.

**Comment by djtorrey** [comment]
it's interesting because it shows in the correct location (top or bottom) when initially running the script, but only upon hitting escape to temporarily hide it does it suddenly go to the center of the screen

**Comment by djtorrey** [comment]
another really odd thing, if I manually set the hres to 1900 or less then it stays on top.  I wonder what the issue of having it at 1920 is.  

For the time being, though, it more or less works on centos so i should be good to go.  I noticed that on centos it wasn't hugging the top and bottom properly but I think that should be ok if I just manually set the resolution.

**Comment by fcaviggia** [comment]
Sorry, I've just never got it working with Ubnutu's display managers - I've pretty much tested it on most other platforms though.

**Comment by InvenTorrey** [comment]
@fcaviggia  I actually think I might have found the issue.  at the top of your code in generating the banners there is a line that sets the position "self.window.set_position(gtk.WIN_POS_CENTER)".  I think that might be causing the issue.  It shouldn't be a necessary line because the banners are being manually positioned to 0,0 (and to the vres for the bottom banner).

I haven't tested it on Ubuntu, but I did some tests with a generic window and it was placing it in the center exactly as ubuntu was placing it after the window was destroyed and brought back.

I also made a couple of other additions to fix some issues I was seeing in centos.  There was a gap at the top and at the bottom with XFCE even though it was being displayed at 0,0.  I actually found that it was impossible for me to try to offset it as well in the negative numbers due to the window manager.  

As a workaround, I used "gtk.WINDOW_POPUP" since popups are apparently not managed by windows managers.  This allowed me to place the window any place I wanted (letting me offset it by 30, which was the XFCE taskbar height).  The same could be done for the bottom.

In addition I added a line so that windows maximized would only maximize to the bottom of the classification bar (so that the classification bar was not overlapping).  The line of code to add is: 

self.window.get_toplevel().show() # must call show() before property_change()
self.window.get_toplevel().window.property_change("_NET_WM_STRUT", 
    "CARDINAL", 32, gtk.gdk.PROP_MODE_REPLACE, [0, 0, 30, 0]) 

(taken from https://stackoverflow.com/questions/3859045/preventing-window-overlap-in-gtk)

this reserves a height of 30 from the top and prevents overlap.  The one thing about this is it doesn't work with popup (negating the previous location fix due to the window manager).  What I did was create another form and instead of using POPUP I used "gtk.WINDOW_TOPLEVEL".  I placed it up top and gave it an opacity of 0.  

It still has some issues.  In other systems if a user full screens it hides the banner, this doesn't do that and it continues to display at an absolute position which could get fixed. I also didn't get around to fixing an issue where escape no longer works, etc... but I see it more as a workaround anyway than an actual implementation.  For now I just turned off escape since it's not needed if everything maximizes below anyway.
